networking/p2p/src/test/java/tech/pegasys/teku/networking/p2p/libp2p/LibP2PPrivateKeyLoaderTest.java

```diff
@@ -62,6 +67,33 @@ void testPrivateKeyLoaded() {
     assertThat(loader.get()).isEqualTo(privKey);
   }
 
+  @Test
+  void shouldCreateKeyAndSaveToFile(@TempDir Path tempDir) throws IOException {
+    final Path file = tempDir.resolve("file.txt");
+    final PrivateKeySource privKeySource = new NetworkConfig.FilePrivateKeySource(file.toString());
+    final LibP2PPrivateKeyLoader loader =
+        new LibP2PPrivateKeyLoader(store, Optional.of(privKeySource));
+
+    final PrivKey key = loader.get();
+    final Bytes generatedBytes = Bytes.fromHexString(Files.readString(file));
+    final PrivKey expectedKey = KeyKt.unmarshalPrivateKey(generatedBytes.toArrayUnsafe());
+    assertThat(key).isEqualTo(expectedKey);
+  }
+
+  @Test
+  void shouldGetKeyFromSavedFile(@TempDir Path tempDir) throws IOException {
+    final Path file = tempDir.resolve("file.txt");
+    final Bytes privateKey = Bytes.wrap(PrivateKeyGenerator.generate().bytes());
+    Files.writeString(file, privateKey.toHexString());
+
+    final PrivateKeySource privKeySource = new NetworkConfig.FilePrivateKeySource(file.toString());
+    final LibP2PPrivateKeyLoader loader =
+        new LibP2PPrivateKeyLoader(store, Optional.of(privKeySource));
+
+    final PrivKey expectedKey = KeyKt.unmarshalPrivateKey(privateKey.toArrayUnsafe());
+    assertThat(loader.get()).isEqualTo(expectedKey);
+  }
+
```

Comment:
These tests are not in the right place... Our tests for `LibP2PPrivateKeyLoader` should not directly depend on `FilePrivateKeySource`.

In this class, we would be testing things like:
1. When `privateKeySource` is empty, we generate a new key
2. When user provides a key (privateKeySource != empty) we use the key provided by the `PrivateKeySource`.

All the tests related to actually creating a file etc should be done in the tests for `FilePrivateKeySource`

Reply:
For reference: see how on test `testPrivateKeyLoaded` we are using a simple implementation of `PrivateKeySource`.

Reply:
Could even use a mock of `PrivateKeySource` to assert when it is used or not.